src/sdk/bucketspace_fs.js

```diff
@@ -203,55 +205,27 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
-        // TODO - replace map & filter to map with concurrency
-        const has_access_buckets = (await P.all(_.map(
-            buckets,
-            async bucket => {
-                dbg.log1('bucketspace_fs.validate_bucket_access:', bucket.name.unwrap());
-                const bucket_config_info = await object_sdk.read_bucket_sdk_config_info(bucket.name.unwrap());
-                const ns = bucket_config_info.namespace;
-                const is_nsfs_bucket = object_sdk.is_nsfs_bucket(ns);
-                const accessible = is_nsfs_bucket ? await this._has_access_to_nsfs_dir(ns, object_sdk) : false;
-                dbg.log1('bucketspace_fs.validate_bucket_access:', bucket.name.unwrap(), is_nsfs_bucket, accessible);
-                return accessible && { ...bucket, creation_date: bucket_config_info.creation_date };
-            }))).filter(bucket_info => bucket_info);
-            return { buckets: has_access_buckets };
-    }
 
-    async _has_access_to_nsfs_dir(ns, object_sdk) {
         const account = object_sdk.requesting_account;
-        dbg.log0('_has_access_to_nsfs_dir: nsr: ', ns, 'account.nsfs_account_config: ', account && account.nsfs_account_config);
-        // nsfs bucket
-        if (!account || !account.nsfs_account_config || _.isUndefined(account.nsfs_account_config.uid) ||
-            _.isUndefined(account.nsfs_account_config.gid)) return false;
-        try {
-            dbg.log0('_has_access_to_nsfs_dir: checking access:', ns.write_resource, account.nsfs_account_config.uid, account.nsfs_account_config.gid);
-            await nb_native().fs.checkAccess({
-                uid: account.nsfs_account_config.uid,
-                gid: account.nsfs_account_config.gid,
-                backend: ns.write_resource.resource.fs_backend,
-                warn_threshold_ms: config.NSFS_WARN_THRESHOLD_MS,
-            }, path.join(ns.write_resource.resource.fs_root_path, ns.write_resource.path || ''));
-
-            return true;
-        } catch (err) {
-            dbg.log0('_has_access_to_nsfs_dir: failed', err);
-            if (err.code === 'ENOENT' || err.code === 'EACCES' || (err.code === 'EPERM' && err.message === 'Operation not permitted')) return false;
-            throw err;
-        }
+        const buckets = await P.map_with_concurrency(10, entries, async entry => {
+            if (native_fs_utils.isDirectory(entry) || !entry.name.endsWith('.json')) {
+                return;
+            }
+            const bucket_name = this.get_bucket_name(entry.name);
+            const bucket = await object_sdk.read_bucket_sdk_config_info(bucket_name);
+            const bucket_policy_accessible = await this.has_bucket_action_permission(bucket, account, 's3:ListBucket');
+            if (!bucket_policy_accessible) return;
+            const fs_accessible = await this.validate_fs_bucket_access(bucket, object_sdk);
+            if (!fs_accessible) return;
+            return bucket;
```

Comment:
Do you know why in the previous implementation `creation_date` was used? (It is not used in your new implementation).

Reply:
yes, it was a part of this fix - https://github.com/noobaa/noobaa-core/pull/7799
bucket contains creation date